src/components/button/_button-base.scss

```diff
@@ -85,9 +85,17 @@ $md-mini-fab-padding: 8px !default;
     @include md-elevation(8);
   }
 
-  &.md-button-focus {
-    background-color: md-color($md-foreground, base, 0.12);
-    @include md-button-theme('background-color', 600);
+  &.md-button-focus:after {
+    // The button spec calls for focus on rasied buttons (and FABs) to be indicated with a
+    // black, 12% opacity shade over the normal color (for both light and dark themes).
+    // We do this by placing an :after psuedo-element with the appropriate shade over the button.
+    position: absolute;
+    top: 0;
+    left: 0;
+    bottom: 0;
+    right: 0;
+    content: '';
+    background-color: rgba(black, 0.12);
```

Comment:
The UX treatments LGTM. However this looks like it only applies to raised buttons. Flat buttons still seem to be [styled using background color](https://github.com/jelbourn/material2/blob/c8462f59f537b8c12573f818f12e021dd6f88411/src/components/button/button.scss#L10). Is the intent to style the two types of buttons using different strategies?